imgee/views/index.py

```diff
@@ -26,215 +20,52 @@ def index():
     return render_template('pop_up_gallery.html', files=files, label=label,
                 profile=profile, uploadform=form, cp_form=cp_form)
 
 
-@app.route('/<profile>/edit_title', methods=['POST'])
-@load_model(Profile, {'name': 'profile'}, 'profile',
-    permission=['edit', 'siteadmin'], addlperms=lastuser.permissions)
-def edit_title(profile):
-    form = forms.EditTitleForm()
-    if form.validate_on_submit():
-        file_name = request.form.get('file_name')
-        q = and_(Profile.userid.in_(_get_owned_ids()), StoredFile.name == file_name)
-        f = StoredFile.query.filter(q).first_or_404()
-        f.title = request.form.get('file_title')
-        db.session.commit()
-        return f.title
-    else:
-        return form.file_title.errors and form.file_title.errors[0], 400
-
-@app.route('/<profile>/<file>/edit_title.json', methods=['POST'])
-@load_models(
-    (Profile, {'name': 'profile'}, 'profile'),
-    (StoredFile, {'name': 'file'}, 'stored_file'),
-    permission=['edit', 'siteadmin'], addlperms=lastuser.permissions)
-def update_title_json(profile, stored_file):
-    form = forms.UpdateTitle()
-    if form.validate_on_submit():
-        old_title = stored_file.title
-        form.populate_obj(stored_file)
-        db.session.commit()
-        return jsonify(status=True, message="Title for %s has been updated to %s" % (old_title, stored_file.title), image_data=stored_file_data(stored_file))
-    else:
-        update_form = render_template('edit_title_form.html', form=form, formid='edit_title_' + stored_file.name)
-        return jsonify(status=False, form=update_form, update_url=url_for('update_title_json', profile=profile.name, file=stored_file.name))
-
-
 @app.route('/<profile>')
 @load_model(Profile, {'name': 'profile'}, 'profile')
 def profile_view(profile):
     files = profile.stored_files.order_by('created_at desc').all()
-    files = not_in_deleteQ(files)
     title_form = forms.EditTitleForm()
     upload_form = forms.UploadImageForm()
     return render_template('profile.html', profile=profile, files=files, uploadform=upload_form, title_form=title_form, mimetypes=ALLOWED_MIMETYPES.keys())
 
 
 @app.route('/<profile>/archive')
 @load_model(Profile, {'name': 'profile'}, 'profile',
-    permission=['view', 'siteadmin'], addlperms=lastuser.permissions)
+    permission=['view', 'siteadmin'])
 def unlabelled_images(profile):
     """Get all unlabelled images owned by profile"""
     files = profile.stored_files.filter(not_(StoredFile.labels.any())).order_by('created_at desc').all()
-    files = not_in_deleteQ(files)
     title_form = forms.EditTitleForm()
     return render_template('profile.html', profile=profile, files=files, title_form=title_form, unlabelled=True)
 
+
 def get_prev_next_images(profile, img, limit=2):
     # query for "all" images though we need just the `limit`
     # bcoz we don't know how many are there in deleteQ.
     imgs = profile.stored_files.order_by('created_at desc').all()
```

Comment:
This query is terribly inefficient. You should ask the database to tell you what's previous and next, and return only those.